File: Parsimmon/Example/TaggerViewController.m
//
//  TaggerViewController.m
//  Example
//
//  Created by Ayaka Nonaka on 10/10/13.
//
//

#import "TaggerViewController.h"
#import "Parsimmon-Swift.h"

@interface TaggerViewController ()
@property (weak, nonatomic) IBOutlet UITextField *inputTextField;
@property (weak, nonatomic) IBOutlet UITextView *outputTextView;

@property (strong, nonatomic) Tagger *tagger;
@end

@implementation TaggerViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
	// Do any additional setup after loading the view, typically from a nib.
    self.tagger = [[Tagger alloc] init];

    UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(dismissKeyboard)];
    [self.view addGestureRecognizer:tap];
}

- (IBAction)parsimmonAction:(id)sender
{
    NSArray *taggedTokens = [self.tagger tagWordsInText:self.inputTextField.text];
    [self.outputTextView setText:[NSString stringWithFormat:@"%@", taggedTokens]];
    [self dismissKeyboard];
}

- (void)dismissKeyboard
{
    [self.view endEditing:YES];
}

@end

File: Parsimmon/Example/TaggerViewController.swift
//
//  ClassifierViewController.swift
//  Parsimmon
//
//  Created by Ayaka Nonaka on 10/10/13.
//
//

import Foundation
import UIKit

class TaggerViewController: UIViewController {
    @IBOutlet private weak var inputTextField: UITextField!
    @IBOutlet private weak var outputTextview: UITextView!
    
    private lazy var tagger = Tagger()
    
    func dismissKeyboard() {
        view.endEditing(true)
    }

    @IBAction private func parsimmonAction(sender: UIButton) {
        let taggedTokens = tagger.tagWordsInText(inputTextField.text)
        outputTextview.text = "\(taggedTokens)"
        dismissKeyboard()
    }
}

extension TaggerViewController {
    override func viewDidLoad() {
        super.viewDidLoad()

        let tap = UITapGestureRecognizer(target: self, action: "dismissKeyboard")
        view.addGestureRecognizer(tap)
    }
}
